updated templates for django 1.9 compatibility (fixes #71)

dsalaj opened the pull request:
I wanted to include django-messages into my project which uses django 1.9. Only compatibility issues I encountered were in templates include `{% load url from future %}`. This is the fix.

pxg commented:
I'm using a fork of this code and it's working well for me with `Django==1.9.4`. I've reviewed the code and it looks good but need to run the test to give this a thumbs up.

I'm a bit confused with what the steps are to run the automated tests. After installing the package I tried `python tests/manage.py test django_messages --settings=tests.settings` but that gives me the output:
```
Traceback (most recent call last):
  File "tests/manage.py", line 17, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/core/management/__init__.py", line 327, in execute
    django.setup()
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/__init__.py", line 18, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/apps/registry.py", line 108, in populate
    app_config.import_models(all_models)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/apps/config.py", line 202, in import_models
    self.models_module = import_module(models_module_name)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "tests/../django_messages/models.py", line 104, in <module>
    from django_messages.utils import new_message_email
  File "tests/../django_messages/utils.py", line 5, in <module>
    from django.contrib.sites.models import Site
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/contrib/sites/models.py", line 83, in <module>
    class Site(models.Model):
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/db/models/base.py", line 102, in __new__
    "INSTALLED_APPS." % (module, name)
RuntimeError: Model class django.contrib.sites.models.Site doesn't declare an explicit app_label and isn't in an application in INSTALLED_APPS.
```

Thanks,
Pete

arneb commented:
@pxg is "django.contrib.sites" in settings.INSTALLED_APPS?

pxg commented:
@arneb no it wasn't. I've added it to `tests/settings.py` and can run the tests now. Thank you.

All the tests fail unless I modify `tests/urls.py` to include the `messages` namespace:
```
from django.conf.urls import include, patterns


urlpatterns = patterns(
    '',
    (r'^messages/', include('django_messages.urls', namespace='messages')),
)
```
With that in place I get `FAILED (failures=1, errors=1)`. I think I know how to fix the tests. Should I do that as a separate PR or is there a way for me to push my fixes to this one?

Full output from tests below:
```
python tests/manage.py test django_messages --settings=tests.settings
Creating test database for alias 'default'...
/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/utils.py:37: RemovedInDjango110Warning: You haven't defined a TEMPLATES setting. You must do so before upgrading to Django 1.10. Otherwise Django will be unable to load templates.
  "unable to load templates.", RemovedInDjango110Warning)

tests/../django_messages/urls.py:16: RemovedInDjango110Warning: django.conf.urls.patterns() is deprecated and will be removed in Django 1.10. Update your urlpatterns to be a list of django.conf.urls.url() instances instead.
  url(r'^trash/$', trash, name='messages_trash'),

/Users/pxg/Projects/dsalaj_django_messages/tests/urls.py:6: RemovedInDjango110Warning: django.conf.urls.patterns() is deprecated and will be removed in Django 1.10. Update your urlpatterns to be a list of django.conf.urls.url() instances instead.
  (r'^messages/', include('django_messages.urls', namespace='messages')),

../Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/shortcuts.py:45: RemovedInDjango110Warning: The context_instance argument of render_to_string is deprecated.
  using=using)

F..E..
======================================================================
ERROR: testReply (django_messages.tests.IntegrationTestCase)
test that user_2 can reply
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/../django_messages/tests.py", line 145, in testReply
    response = self.c.get(reverse('messages:messages_inbox'))
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/test/client.py", line 503, in get
    **extra)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/test/client.py", line 304, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/test/client.py", line 380, in generic
    return self.request(**r)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/test/client.py", line 467, in request
    six.reraise(*exc_info)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/views/decorators/cache.py", line 57, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/contrib/auth/decorators.py", line 23, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "tests/../django_messages/views.py", line 34, in inbox
    }, context_instance=RequestContext(request))
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/shortcuts.py", line 45, in render_to_response
    using=using)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/loader.py", line 114, in render_to_string
    template_name, context, context_instance, dirs, dictionary)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/engine.py", line 244, in render_to_string
    return t.render(context_instance)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 206, in render
    return self._render(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/test/utils.py", line 92, in instrumented_test_render
    return self.nodelist.render(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/loader_tags.py", line 173, in render
    return compiled_parent._render(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/test/utils.py", line 92, in instrumented_test_render
    return self.nodelist.render(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/loader_tags.py", line 173, in render
    return compiled_parent._render(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/test/utils.py", line 92, in instrumented_test_render
    return self.nodelist.render(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/loader_tags.py", line 69, in render
    result = block.nodelist.render(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/defaulttags.py", line 326, in render
    return nodelist.render(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/defaulttags.py", line 220, in render
    nodelist.append(node.render_annotated(context))
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 1043, in render
    output = self.filter_expression.resolve(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 709, in resolve
    obj = self.var.resolve(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 850, in resolve
    value = self._resolve_lookup(context)
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/template/base.py", line 913, in _resolve_lookup
    current = current()
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/db/models/__init__.py", line 43, in inner
    return reverse(bits[0], None, *bits[1:3])
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/core/urlresolvers.py", line 600, in reverse
    return force_text(iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs)))
  File "/Users/pxg/.virtualenvs/django_messages/lib/python2.7/site-packages/django/core/urlresolvers.py", line 508, in _reverse_with_prefix
    (lookup_view_s, args, kwargs, len(patterns), patterns))
NoReverseMatch: Reverse for 'messages_detail' with arguments '(1,)' and keyword arguments '{}' not found. 0 pattern(s) tried: []

======================================================================
FAIL: testCompose (django_messages.tests.IntegrationTestCase)
compose a message step by step
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/../django_messages/tests.py", line 129, in testCompose
    "http://testserver%s" % reverse('messages:messages_inbox'))
AssertionError: '/messages/inbox/' != u'http://testserver/messages/inbox/'

----------------------------------------------------------------------
Ran 8 tests in 2.267s

FAILED (failures=1, errors=1)
```

dsalaj commented:
@pxg I have added the ´messages´ namespace explicitly to this branch because I needed it in my project (I was not able to figure out how to add namespace without breaking the `url` tag in templates).

I have added you to collaborators in my fork, you can push now directly.

pxg commented:
@dsalaj thanks for the access I have fixed the tests. Are you happy for me to push to master on https://github.com/dsalaj/django-messages or would you like me to use a different branch?

pxg commented:
@arneb what's your opinion on the url namespace changes? It seems a good idea to me, from reading the Django docs https://docs.djangoproject.com/es/1.9/topics/http/urls/#url-namespaces "It's a good practice for third-party apps to always use namespaced URLs".

dsalaj commented:
@pxg Feel free to push to master. And thanks for the link to docs on the namespace issue. I am also interested in the opinion of @arneb. Was there a way to avoid my changes in reverse calls (see [this commit](https://github.com/arneb/django-messages/pull/73/commits/5447b4722cc02ba0a4c228d7c92c22c9bebfbd2e)) and still add a namespace to this app from project when using this app?

luvfknhrts commented on django_messages/templates/django_messages/inbox.html:
```diff
@@ -1,16 +1,15 @@
         <tr><th>{% trans "Sender" %}</th><th>{% trans "Subject" %}</th><th>{% trans "Received" %}</th><th>{% trans "Action" %}</th></tr>
     </thead>
     <tbody>
-{% for message in message_list %} 
```
13

luvfknhrts commented on django_messages/forms.py:
```diff
@@ -10,6 +10,7 @@
 
```
Import

arneb commented:
thank you, but it is no longer needed, the current code is now compatible
